common/test/acceptance/pages/studio/settings.py

```diff
@@ -61,8 +63,13 @@ def save_changes(self, wait_for_confirmation=True):
                 'Waiting for save confirmation...'
             ).fulfill()
 
-    def refresh_page(self):
+    def refresh_page(self, wait_for_confirmation=True):
         """
         Reload the page.
         """
         self.browser.refresh()
+        if wait_for_confirmation:
+            EmptyPromise(
+                lambda: self.q(css='body.view-settings').present,
+                'Waiting for page refresh to complete...'
```

Comment:
The description should be a description of the promise. If it fails the log message will say that the promise was broken. So here you shouldn't say 'Waiting for page refresh to complete...' but instead say something like 'Page refresh has completed.'

http://bok-choy.readthedocs.org/en/latest/api_reference.html?highlight=promise#bok_choy.promise.Promise